### PR TITLE
Fix e2e plugin sync tmc test

### PR DESCRIPTION
**Pull request by mpanchajanya**
### What this PR does / why we need it
- Remove the corrupt validation function
### Which issue(s) this PR fixes

Fixes #

### Describe testing done for PR

`
make e2e-plugin-sync-tmc

Ginkgo ran 1 suite in 3m38.904802833s
Test Suite Passed
`

### Release note

```release-note

```

### CODE REVIEW

File: test/e2e/plugin_sync/tmc/plugin_sync_tmc_lifecycle_test.go [hunk 1]
```diff
@@ -605,15 +604,38 @@ var _ = f.CLICoreDescribe("[Tests:E2E][Feature:Plugin-Sync-TMC-lifecycle]", func
 			Expect(f.CheckAllPluginsExists(installedPluginsListTMC, pluginsToGenerateMockResponse)).Should(BeTrue(), pluginsInstalledAndMockedShouldBeSame)
 		})
 
-		// Test case: h. validate plugin list consistancy, it should sort by context name always
+		// Test case: h. validate plugin list consistency, it should sort by context name always
```
Comment (marckhouzam) [inline test/e2e/plugin_sync/tmc/plugin_sync_tmc_lifecycle_test.go:607]: "it should sort by context name always" this is no longer what we check, right?

File: test/e2e/plugin_sync/tmc/plugin_sync_tmc_lifecycle_test.go [hunk 2]
```diff
@@ -605,15 +604,38 @@ var _ = f.CLICoreDescribe("[Tests:E2E][Feature:Plugin-Sync-TMC-lifecycle]", func
...
 				Expect(err).To(BeNil(), noErrorForPluginList)
 				Expect(totalInstalledPlugins).Should(Equal(len(installedPlugins)), "total installed plugins count should be equal to plugins installed for both contexts")
-				sort.Strings(contexts)
-				Expect(f.ValidateInstalledPluginsOrder(contexts, installedPlugins)).To(BeTrue())
+
+				// Filter plugins without context
+				var installedPluginsWithContext []*f.PluginInfo
+				for _, plugin := range installedPlugins {
+					if plugin.Context != "" {
+						installedPluginsWithContext = append(installedPluginsWithContext, plugin)
```
Comment (marckhouzam) [inline test/e2e/plugin_sync/tmc/plugin_sync_tmc_lifecycle_test.go:635]: this array will contain the plugins from both contexts right?  k8s and tmc.
Those are not sorted alphabetically, but by context, then alphabetically.  I don't understand how the test is passing (although I don't know what the plugins are named, so maybe we are lucky and they are all alphabetical anyway?)

Do you understand what is happening?

Comment (mpanchajanya) [reply test/e2e/plugin_sync/tmc/plugin_sync_tmc_lifecycle_test.go:635]: @marckhouzam  Originally I thought the same but it looks like the plugin list data is sorted by name only. 
This is the response of plugin name and plugin context
```
[i] Executing command: tanzu plugin list -o json
[account apply audit cluster feature kubernetes-release]
[plugin-sync-tmc-FVYj plugin-sync-tmc-FVYj plugin-sync-tmc-FVYj plugin-sync-k8s-UZCH plugin-sync-k8s-UZCH plugin-sync-k8s-UZCH]
```

Comment (marckhouzam) [reply test/e2e/plugin_sync/tmc/plugin_sync_tmc_lifecycle_test.go:635]: Oh wow, you are right!  Sorry about that.

**Review by marckhouzam — APPROVED** [review]
LGTM
Minor nit about a comment

**Pull request merged**